operator/v1alpha1/types_csi.go

```diff
@@ -0,0 +1,69 @@
+package v1alpha1
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterCSIDriver object allows management and configuration of a CSI driver operator
+// installed by default in OpenShift.
+type ClusterCSIDriver struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ClusterCSIDriverSpec `json:"spec"`
+
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status ClusterCSIDriverStatus `json:"status"`
+}
+
+// CSIDriverName is the name of the CSI driver
+type CSIDriverName string
+
+// If you are adding a new driver name here, ensure that kubebuilder:validation:Enum below is updated
+// and 0000_20_cluster_csi_driver.crd.yaml-merge-patch file is also updated with new driver name.
+const (
+	AWSEBSCSIDriver CSIDriverName = "ebs.csi.aws.com-ebs"
+	ManilaCSIDriver CSIDriverName = "manila.csi.openstack.org"
+	OvirtCSIDriver  CSIDriverName = "csi.ovirt.org"
+)
+
+// ClusterCSIDriverSpec is the desired behavior of CSI driver operator
+type ClusterCSIDriverSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+	// +kubebuilder:validation:Required
+	// +required
+	DriverConfig CSIDriverConfig `json:"driverConfig"`
+}
+
+// ClusterCSIDriverStatus is the observed status of CSI driver operator
+type ClusterCSIDriverStatus struct {
+	operatorv1.OperatorStatus `json:",inline"`
+}
+
+// CSIDriverConfig is the CSI driver specific configuration
+type CSIDriverConfig struct {
+	// DriverName holds the name of the CSI driver
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=ebs.csi.aws.com-ebs;manila.csi.openstack.org;csi.ovirt.org
+	// +required
+	DriverName CSIDriverName `json:"driverName"`
+}
```

Comment:
Add some comments indicating an example of what a driver specific configuration would look like inside here.

Reply:
added.